src/main/java/de/thetaphi/forbiddenapis/Checker.java

```diff
@@ -419,8 +424,23 @@ private int checkClass(final ClassReader reader, Pattern suppressAnnotationsPatt
       }
       msg.append(": ").append(cause);
       msg.append(" (while looking up details about referenced class '").append(rcle.getClassName()).append("')");
-      assert cause != null && (cause instanceof IOException || cause instanceof ClassNotFoundException);
+      assert cause != null && (cause instanceof IOException || cause instanceof ClassNotFoundException || cause instanceof RuntimeException);
       throw new ForbiddenApiException(msg.toString(), cause);
+    } catch (RuntimeException re) {
+      if (AsmUtils.isExceptionInAsmClassReader(re)) {
+        final StringBuilder msg = new StringBuilder()
+            .append("Failed to parse class '")
+            .append(className)
+            .append('\'');
+        final String source = scanner.getSourceFile();
+        if (source != null) {
+          msg.append(" (").append(source).append(')');
+        }
+        msg.append(": ").append(re);
+        throw new ForbiddenApiException(msg.toString(), re);
+      }
+      // else rethrow (it's occuring in our code):
```

Comment:
@vlsi here could be some logging information emmitted where it actually parses the file, to help catching bugs in forbiddenapis itsself. Class file and line numbers.

Reply:
I opened an issue about the additional "parsing state" printed to error logger in case of exception: #175